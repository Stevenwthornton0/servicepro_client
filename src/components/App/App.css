
.red {
    color: red;
}

@media screen and (min-width: 500px) {
    .mobile {
        display: none;
    }
}

@media screen and (max-width: 500px) {
    .computer {
        display: none;
    }
}