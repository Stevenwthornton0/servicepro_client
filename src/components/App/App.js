import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from '../Header/Header';
import PrivateOnlyRoute from '../../Utils/PrivateOnlyRoute';
import PublicOnlyRoute from '../../Utils/PublicOnlyRoute';
import ProviderRegistration from '../../Routes/ProviderRegistration/ProviderRegistration';
import HomePage from '../../Routes/HomePage/HomePage';
import Services from '../../Routes/Services/Services';
import ServiceList from '../../Routes/ServiceList/ServiceList';
import LoginPage from '../../Routes/LoginPage/LoginPage';
import ServicePage from '../../Routes/ServicePage/ServicePage';
import EditPage from '../../Routes/EditPage/EditPage';
import UserRegistration from '../../Routes/UserRegistration/UserRegistration';
import './App.css';

class App extends Component {
  state = { hasError: false }

  static getDerivedStateFromError(error) {
    console.log(error)
    return { hasError: true }
  }

  render() {
    return (
      <div className='App'>
        
        <header className='header'>
          <Header />
        </header>

        <main className='App_main'>
          <Switch>

            <Route 
              exact
              path={'/'}
              component={HomePage}
            />

            <PrivateOnlyRoute
              path={'/provider/registration'}
              component={ProviderRegistration}
            />

            <PrivateOnlyRoute
              exact
              path={'/services'}
              component={Services}
            />

            <PrivateOnlyRoute
              exact
              path={'/services/:serviceType'}
              component={ServiceList}
            />

            <PrivateOnlyRoute 
              exact
              path={'/services/service/:serviceId'}
              component={ServicePage}
            />

            <PrivateOnlyRoute 
              exact
              path={'/services/service/:serviceId/edit'}
              component={EditPage}
            />

            <PublicOnlyRoute 
              exact
              path={'/register'}
              component={UserRegistration}
            />

            <PublicOnlyRoute 
              path={'/login'}
              component={LoginPage}
            />

          </Switch>
        </main>

      </div>
    )
  }
}

export default App;