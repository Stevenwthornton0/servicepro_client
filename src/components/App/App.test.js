import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import App from './App';

Enzyme.configure({ adapter: new Adapter() })

describe('App component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<App />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})