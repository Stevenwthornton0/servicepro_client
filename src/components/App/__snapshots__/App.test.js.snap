// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <Header />
  </header>
  <main
    className="App_main"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <PrivateOnlyRoute
        component={[Function]}
        path="/provider/registration"
      />
      <PrivateOnlyRoute
        component={[Function]}
        exact={true}
        path="/services"
      />
      <PrivateOnlyRoute
        component={[Function]}
        exact={true}
        path="/services/:serviceType"
      />
      <PrivateOnlyRoute
        component={[Function]}
        exact={true}
        path="/services/service/:serviceId"
      />
      <PrivateOnlyRoute
        component={[Function]}
        exact={true}
        path="/services/service/:serviceId/edit"
      />
      <PublicOnlyRoute
        component={[Function]}
        exact={true}
        path="/register"
      />
      <PublicOnlyRoute
        component={[Function]}
        path="/login"
      />
    </Switch>
  </main>
</div>
`;
